### ISSUE TITLE
API key authentication method does not work with Xray resources

### ISSUE THREAD

**Issue by drcrees** [jfrog/terraform-provider-artifactory#106]
**Describe the bug**

When using the following auth method,
```hcl
provider "artifactory" {
  url          = "https://artifactory.company.cloud"
  api_key = "<API_KEY>"
}
```
we found ourselves unable to create Xray policies.

```hcl
# Create a new Xray license policy
resource "artifactory_xray_policy" "block_download_high_severity" {
  name  = "block-download-high-severity"
  type  = "security"
  rules {
    name     = "rule1"
    priority = 1
    criteria {
      min_severity = "High"
    }
    actions {
      block_download {
        unscanned = true
        active    = true
      }
    }
  }
}
```

Username and password method works as expected.
The errors surfaced were not too helpful in determining cause.

Info:
```text
Error:
```
Trace:
```text
Error: rpc error: code = Unavailable desc = transport is closing
```

Artifactory reported a `403`.

This may be an issue with the underlying Xray client.

**Requirements for and issue**
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [x] Your version of terraform

Artifactory version 7.18.3
Xray version 3.23.0
Terraform version 1.0.4

**Expected behavior**
I expected to be able to create an Xray policy using API key auth method, received non-descriptive error instead.
Switching the auth method to username and password resolved the issue.

**Comment by chb0github** [comment]
What version of the provider are you using? I will have a look

**Comment by drcrees** [comment]
v2.2.15

**Comment by chb0github** [comment]
To be clear: when you say API_KEY you mean the one in the UI under:
http://localhost:8082/ui/admin/artifactory/user_profile

and not the one under
http://localhost:8082/ui/admin/configuration/security/access_tokens

**Comment by drcrees** [comment]
@chb0github correct, it is a user API key.  We have a service account user specifically for running terraform.

`api_key` auth method has worked fine up until we tried to create a `artifactory_xray_policy`

**Comment by chb0github** [comment]
Ok, So, I do remember this issue, and having a discussion with the XRAY team about it. At some point the decision was made to only allow AUTH_TOKENS for their APIs. 

There is, however a workaround that shouldn't be too much bother. Apparently you can use the `auth_token` as the password for use in username/pass auth.

Let me demonstrate:

attempt to access xray policies using API key results in 403:
```bash
christianb@christianb-mac terraform-provider-artifactory % curl -w '%{http_code}' -H "X-JFrog-Art-Api: AKCp..." -sL https://partnership.jfrog.io/xray/api/v1/policies 
403%   
```
Attempting the same exact thing, except against an artifactory endpoint, works fine:

```bash
curl -H "X-JFrog-Art-Api: AKCp..."-sL https://partnership.jfrog.io/artifactory/api/repositories | jq 'length'      
76
```
Now, let's try the exact same XRAY URL where we use the username as normal, but the password is the auth token:
```bash
curl -u "christianb:AKCp..." -sL https://partnership.jfrog.io/xray/api/v1/policies | jq length
18
```

I wouldn't blame you for slapping your forehead on this. You're welcome to cut a ticket to the XRAY team through support channels. 

Bottom line:
Use the username and password option where
username: is the user that has the api key
password: is the api key
